In [ ]:
# Coding Discussion 2
## Charlie Zhang

In [2]:
## Read in the data (we would provide this)
import csv
with open('us-states.csv') as file:
        state_covid_data = []
        for row in csv.reader(file):
            state_covid_data.append(row)

len(state_covid_data)

10080

In [3]:
state_covid_data[:5]

[['date', 'state', 'fips', 'cases', 'deaths'],
 ['2020-01-21', 'Washington', '53', '1', '0'],
 ['2020-01-22', 'Washington', '53', '1', '0'],
 ['2020-01-23', 'Washington', '53', '1', '0'],
 ['2020-01-24', 'Illinois', '17', '1', '0']]

### (1) Count up the number of unique dates in the data.

In [4]:
## Create a new list that contains date only 
date = []

## Use for loop to add the date within each list (skip the first list)
for item in state_covid_data[1:]:
    date.extend(item[:1])

print(len(set(date)))

225


There are 225 unique dates in the data.

### (2) Find the first date in which the District of Columbia recorded a case.

In [5]:
## Create a new list that contains state only
state= []

## Use for loop to add the state within each list (skip the first list)
for item in state_covid_data[1:]:
    state.extend(item[1:2])
    
## Use list.index to return the first case in DC (considering skip the first list)
dc_firstday = state.index("District of Columbia") + 1

## Use the index to find the first day in the original list
print("The first date in which DC recorded a case is "+ state_covid_data[dc_firstday][0] + ".")

The first date in which DC recorded a case is 2020-03-07.


### (3) Write a function that takes in a state name as input (e.g. "Wisconsin") and outputs the date of its first case.

In [6]:
## Write a function named *lookupdate*

def lookupdate(statename): 
    state= []
    for item in state_covid_data[1:]: 
        state.extend(item[1:2])
    state_firstday = state.index(statename) + 1
    return print(state_covid_data[state_firstday][0] + " is the date of first case in " + statename + ".")

lookupdate("Wisconsin")

2020-02-05 is the date of first case in Wisconsin.


### (Optional) Write a function that takes in a state name as input (e.g. "Wisconsin") and outputs the date when the number of reported cases within the state exceeded 1000.

In [6]:
def exceed1000(statename): 
   ## Create a list named *matches* that fulfills both conditions
   matches = [s for s in state_covid_data if s[1]== statename and int(s[3])>=1000]
   return print(matches[0][0] + " is the date when the numbers of reported cases within the state exceeded 1000")

## Run a test for Wisconsin
exceed1000("Wisconsin")

2020-03-28 is the date when the numbers of reported cases within the state exceeded 1000
